src/sentry/api/endpoints/data_export.py

```diff
@@ -32,9 +33,14 @@ def post(self, request, organization):
         if not features.has("organizations:data-export", organization):
             return Response(status=404)
 
+        limit = request.data.get("limit")
         serializer = ExportedDataSerializer(
             data=request.data, context={"organization": organization, "user": request.user}
         )
+        try:
+            environment_id = self._get_environment_id_from_request(request, organization.id)
+        except Environment.DoesNotExist as error:
+            return Response(error, status=400)
```

Comment:
So this is how the `environment_id` is fetched in the existing export code, so I just moved it over to Async export, but I couldn't manage to get it testing properly locally, so it I've never been able to have an `enviornment_id` that isn't `None`

Reply:
I think a `404` might be more appropriate if you can't find the environment